### PR TITLE
Add codeql check and CVE scanning

### PR DESCRIPTION
**Pull request by yaocw2020**

### CODE REVIEW

**Review by guangbochen — COMMENTED** [review]
can u please also bump the golangci and dapper base image to BCI golang. e.g., https://github.com/harvester/network-controller-harvester/blob/master/Dockerfile.dapper#L1-L10

File: .github/workflows/scan.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: trivy-scanning
+on:
+  push:
+    branches:
+      - master
+      - 'v*'
```
Comment (guangbochen) [inline .github/workflows/scan.yml:6]:
```suggestion
      - 'v**'
```

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by chrisho — APPROVED** [review]
LGTM, thanks

**Pull request merged**